src/index.js

```diff
@@ -0,0 +1,223 @@
+  } else if (state.currentTemp >= 50) {
+    tempValueElement.style.color = "green";
+    landscapeElement.textContent = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+  } else if (state.currentTemp <= 49) {
+    tempValueElement.style.color = "blue";
+    landscapeElement.textContent = "⛄️ ⛄️ ⛄️";
+  }
+};
+
+// Function to increase temperature
+const increaseTemp = () => {
+  state.currentTemp += 1;
+  updateTemperatureDisplay();
+};
+
+// Function to decrease temperature
+const decreaseTemp = () => {
+  state.currentTemp -= 1;
+  updateTemperatureDisplay();
+};
+
+/** Wave 2 events **/
+const temperatureControls = () => {
+  const warmUpButton = document.getElementById("increaseTempControl");
+  const coolDownButton = document.getElementById("decreaseTempControl");
+
+  warmUpButton.addEventListener("click", increaseTemp);
+  coolDownButton.addEventListener("click", decreaseTemp);
+};
+
+
+/************************/
+/******* Wave 3 *********/
+/************************/
+
+const updateCity = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const currentCityHeader = document.getElementById("headerCityName");
+  currentCityHeader.textContent = cityNameInput.value;
+};
+
+const handleUpdateCityEvent = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const resetButton = document.getElementById("cityNameReset");
+  cityNameInput.addEventListener("input", updateCity);
+  resetButton.addEventListener("click", ResetCity);
+};
+  
+/************************/
+/******* Wave 4 *********/
+/************************/
+
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+// Event handler function to retrieve user input:
+const retrieveCityInput = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  return cityNameInput.value;
+};
+
+// API Call to LocationIQ for location coordinates
+const getCoordinates = () => {
+  let location = retrieveCityInput();
+
+  return axios
+    .get("http://127.0.0.1:5000/location", {
+      params: {
+        q: location,
+      },
+    })
+    .then((response) => {
+      let latitude = response.data[0].lat;
+      let longitude = response.data[0].lon;
+
+      console.log(
+        `Location: ${location}, Latitude: ${latitude}, Longitude: ${longitude}`
+      );
+      return {
+        latitude: latitude,
+        longitude: longitude,
+      };
+    })
+    .catch((error) => {
+      console.log("Error found inside getCoordinates function!");
+      console.log(
+        `The value of status inside of error response is: 
+        ${error.response.status}`
+      );
+    });
+};
+
+// API Call for weather status/temp of current city displayed
+const getCurrentCityWeather = () => {
+  // Info from OpenWeather API Call documentation:
+  // https://openweathermap.org/current#geo
+  return getCoordinates()
+    .then((coordinatesResponse) => {
+      if (!coordinatesResponse) {
+        throw new Error("Coordinates could not be retrieved.");
```

Comment:
You could also throw this error in the catch above so that the error doesn't propagate further down like it does here.